src/System.CommandLine.DragonFruit/CommandLine.cs

```diff
@@ -133,10 +133,17 @@ public static class CommandLine
             if (method.GetParameters()
                       .FirstOrDefault(p => _argumentParameterNames.Contains(p.Name)) is ParameterInfo argsParam)
             {
+                IArgumentArity arity = null; // default
+                if (!argsParam.ParameterType.IsArray && argsParam.HasDefaultValue)
```

Comment:
Have a look at `ArgumentArity.Default` which has some heuristics for deriving arity. That method could be generalized into something like this and made public:

```cs
public IArgumentArity Default(Type type, bool isCommand, bool hasDefaultValue)
```

Reply:
Another approach that might work is to call `SetDefaultValue(null)` on the argument when `argsParam.HasDefaultValue == true`.